src/modules/px4iofirmware/mixer.cpp

```diff
@@ -287,19 +295,29 @@ mixer_tick(void)
 		/* mix */
 		mixed = mixer_mix_threadsafe(&outputs[0], &r_mixer_limits);
 
-		/* the pwm limit call takes care of out of band errors */
-		pwm_limit_calc(should_arm, should_arm_nothrottle, mixed, r_setup_pwm_reverse, r_page_servo_disarmed,
-			       r_page_servo_control_min, r_page_servo_control_max, outputs, r_page_servos, &pwm_limit);
+		/* if mixer_mix_threadsafe returns zero, it did nothing */
+		if (mixed != 0) {
```

Comment:
Let me check.

For multirotors, it's `_rotor_count` https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/mixer/mixer_multirotor.cpp#L421. It's set using the underlying geometry, so it should be something valid if the mixer was loaded correctly and wouldn't just be erratic during flight.

Also for fixedwing (simple mixer) it looks straightforward: https://github.com/PX4/Firmware/blob/e7db0ed098bcc030042aba0667d09da67c130ffe/src/modules/systemlib/mixer/mixer_simple.cpp#L313.

Therefore, I think the implementation as it was is fine.